### PR TITLE
Test clone with target pvc name conflict

### PR DESCRIPTION
**Pull request by brybacki**
Signed-off-by: Bartosz Rybacki <brybacki@redhat.com>

**What this PR does / why we need it**:

Adds a new test case to verify correct behavior when the target pvc name is taken.

PVC name conflict is already verified once by validation hook on dv creation. This tests only covers a situation when the target pvc name is taken between the dv creation and the actual clone process starting.  

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by awels — COMMENTED** [review]
So I am missing something here, if I read the test correctly we do this:
1. Make source
2. Have a pod use the source.
3. Start a clone
4. Since 2 is using the source, the clone won't proceed
5. Make a different PVC with the target name in the target namespace
6. Check for events that the resource exists.

The thing I am missing is that we never delete the pod we created in 2, so we never actually attempt to do the clone. So where are the events coming from?

**Comment by brybacki** [comment]
> So I am missing something here, if I read the test correctly we do this:
> 
> 1. Make source
> 2. Have a pod use the source.
> 3. Start a clone
> 4. Since 2 is using the source, the clone won't proceed
> 5. Make a different PVC with the target name in the target namespace
> 6. Check for events that the resource exists.
> 
> The thing I am missing is that we never delete the pod we created in 2, so we never actually attempt to do the clone. So where are the events coming from?

This is the only way I could create a target name conflict in a deterministic way. 

First event is obviously a source in use: `verifyEvent(controller.SmartCloneSourceInUse, targetNamespaceName, f)`. Then the pod is deleted:
https://github.com/kubevirt/containerized-data-importer/blob/db4df4eb348ca2ea0ee876942b24f5abcfba7c19/tests/cloner_test.go#L246
and 253
```					err = f.K8sClient.CoreV1().Pods(f.Namespace.Name).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
					Expect(err).ToNot(HaveOccurred())
```

And then clone can proceed.

**Review by kgoldbla — APPROVED** [review]
CNV-6612 is covered in this PR. Looks good

**Comment by awels** [comment]
Oh I been reviewing too much, I missed the delete completely. Now it makes a lot more sense.

File: tests/cloner_test.go [hunk 1]
```diff
@@ -208,6 +208,56 @@ var _ = Describe("all clone tests", func() {
...
+
+				targetPvc, err := utils.CreatePVCFromDefinition(f.K8sClient, targetNamespaceName,
+					utils.NewPVCDefinition("target-pvc", "1Gi", annotations, nil))
+				Expect(err).ToNot(HaveOccurred())
+				f.ForceBindIfWaitForFirstConsumer(targetPvc)
+
+				actualCloneType := utils.GetCloneType(f.CdiClient, dataVolume)
+				if actualCloneType == "snapshot" {
+					verifyEvent(controller.SmartCloneSourceInUse, targetNamespaceName, f)
+					err = f.K8sClient.CoreV1().Pods(f.Namespace.Name).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
```
Comment (awels) [inline tests/cloner_test.go:246]: this line to line 250 is duplicated on line 253-257. In order to keep with the DRY principle can we do
```go
if actualCloneType == "snapshot" {
  verifyEvent(controller.SmartCloneSourceInUse, targetNamespaceName, f)
} else if actualCloneType == "csiclone" {
  verifyEvent(controller.CSICloneSourceInUse, targetNamespaceName, f)
} else {
  Fail(fmt.Sprintf("Unknown clonetype %s", actualCloneType))
}
err = f.K8sClient.CoreV1().Pods(f.Namespace.Name).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
Expect(err).ToNot(HaveOccurred())
//verify event
verifyEvent(controller.ErrResourceExists, targetNamespaceName, f)
```
And as a bonus we now catch if for some reason the clone type gets corrupted.

Comment (brybacki) [reply tests/cloner_test.go:246]: thanks, that is good, applied

**Comment by brybacki** [comment]
@awels so wdyt?

**Pull request merged**